dhcpv4/nclient4/client.go

```diff
@@ -225,6 +260,24 @@ func WithTimeout(d time.Duration) ClientOpt {
 	}
 }
 
+// WithSummaryLogger logs one-line DHCPv4 message summarys when sent & received.
+func WithSummaryLogger() ClientOpt {
+	return func(c *Client) {
+		c.logger = shortSummaryLogger{
+			Logger: log.New(os.Stderr, "[dhcpv4]", log.LstdFlags),
+		}
+	}
+}
+
+// WithDebugLogger logs multi-line full DHCPv4 messages when sent & received.
+func WithDebugLogger() ClientOpt {
```

Comment:
You could add another option which takes an exported logger interface containing only Printf so that users can pass in e.g. a custom log.Logger.